src/Workspaces/Core/MSBuild/Workspaces.MSBuild.csproj

```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
+<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\..\..\..\build\Targets\SettingsSdk.props" />
+  <PropertyGroup>
+    <Platform Condition="'$(Platform)' == ''">AnyCPU</Platform>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <OutputType>Library</OutputType>
+    <RootNamespace>Microsoft.CodeAnalysis</RootNamespace>
+    <AssemblyName>Microsoft.CodeAnalysis.Workspaces.MSBuild</AssemblyName>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <TargetFramework>net46</TargetFramework>
```

Comment:
:memo: I have a branch locally where the goal was making this `netstandard1.5` (the lowest standard which the MSBuild APIs support). It was mostly complete but there was a section of important code which couldn't be purely extracted from Workspaces.Desktop using strictly type forwarding to preserve compatibility so I didn't push it forward.

Reply:
Ultimately, the plan is to make this netstandard2.0.

Reply:
Cool, that wasn't an option at the time I looked so here's to hoping netstandard2.0 has all the APIs to make it straightforward success. 🍻

Reply:
To be honest, it's not _quite_ an option for this yet either. Moving to netstandard2.0 has implications for Visual Studio, though that might be a good thing to sort through and address now. :smile:

Reply:
:memo: It will be a big deal for a long-standing request to provide command line automation for analyzers/fixes.

Reply:
yup.

Reply:
Do we ever load this assembly in Visual Studio though?

Reply:
_Maybe..._

Microsoft.CodeAnalysis.Workspaces.Desktop ships in VS today. It's possible that VS extensions may have targeted it.

Reply:
What is the use case for this particular library? Is it only ever shipped via NuGet? That will impact our netstandard discussions. 

---
In reply to: [156729257](https://github.com/dotnet/roslyn/pull/21670#discussion_r156729257) [](ancestors = 156729257)